src/requests/textdocument.jl

```diff
@@ -256,19 +254,102 @@ function edit_string(text, editrange, edit)
+    errs = StaticLint.collect_hints(getcst(doc))
+    n = length(errs)
+    n == 0 && return out
+    i = 1
+    start = true
+    offset = errs[i][1]
+    
+    r = Int[0, 0]
+    pos = 0
+    nlines = length(line_offsets)
+    if offset > last(line_offsets)
+        line = nlines
+    else
+        line = 1
+        while line < nlines
+            while line_offsets[line] <= offset < line_offsets[line + 1]
+                ind = line_offsets[line]
+                char = 0
+                while offset > ind
+                    ind = nextind(doc._content, ind)
+                    char += 1
+                end                
+                if start
+                    r[1] = line
+                    r[2] = char
+                    offset += errs[i][2].span
+                else
+                    if typof(errs[i][2]) === CSTParser.ErrorToken
+                        push!(out, Diagnostic(Range(r[1] - 1, r[2], line - 1, char), 1, "Julia", "Julia", "Parsing error", nothing))
+                    elseif typof(errs[i][2]) === CSTParser.IDENTIFIER
+                        push!(out, Diagnostic(Range(r[1] - 1, r[2], line - 1, char), 2, "Julia", "Julia", "Missing reference: $(errs[i][2].val)", nothing))
+                    elseif StaticLint.haserror(errs[i][2]) && StaticLint.errorof(errs[i][2]) isa StaticLint.LintCodes
+                        push!(out, Diagnostic(Range(r[1] - 1, r[2], line - 1, char), 3, "Julia", "Julia", get(StaticLint.LintCodeDescriptions, StaticLint.errorof(errs[i][2]), ""), nothing))
+                    end
+                    i += 1
+                    i>n && break
+                    offset = errs[i][1]
+                end
+                start = !start
+                offset = start ? errs[i][1] : errs[i][1] + errs[i][2].span
+            end
+            line += 1
         end
     end
-    binds, refs, urefs
+    return out
+end
+
+
+function convert_diagnostic(h::LSDiagnostic{T}, doc::Document) where {T}
+    rng = Range(doc, h.loc)
+    code = 1
+    Diagnostic(rng, code, string(T), string(typeof(h).name), isempty(h.message) ? string(T) : h.message, nothing)
+end
+
+function publish_diagnostics(doc::Document, server)
+    ls_diags = map(doc.diagnostics) do diag
+        convert_diagnostic(diag, doc)
+    end
+    publishDiagnosticsParams = PublishDiagnosticsParams(doc._uri, ls_diags)
+    response =  JSONRPC.Request{Val{Symbol("textDocument/publishDiagnostics")},PublishDiagnosticsParams}(nothing, publishDiagnosticsParams)
+    send(response, server)
+end
+
+
+function clear_diagnostics(uri::URI2, server)
+    doc = server.documents[uri]
+    empty!(doc.diagnostics)
+    response =  JSONRPC.Request{Val{Symbol("textDocument/publishDiagnostics")},PublishDiagnosticsParams}(nothing, PublishDiagnosticsParams(doc._uri, Diagnostic[]))
+    send(response, server)
+end 
+
+function clear_diagnostics(server)
+    for (uri, doc) in server.documents
+        clear_diagnostics(uri, server)
+    end
```

Comment:
Are the changes around `publish_diangostics` etc on purpose? They undo https://github.com/julia-vscode/LanguageServer.jl/pull/423, which fixed the configuration handling for those cases.

Reply:
Also, I removed the type `LSDiagnostic`, but here it is used again, so this probably also just doesn't work as is?

Reply:
Nope was an error, reverted